### PR TITLE
share x post bottom sheet on mobile

### PR DESCRIPTION
**Pull request by Rohan-cp**
### Summary of Changes
brings up a bottom sheet to share a post where you can copy the post link. From this bottom sheet, you can also share on x platform with a prepopulated message that includes the post link

all the different ways to trigger it-
1. shows up when you successfully post a token
2. click on '...' on feedPost and then click "Share"

### Demo 

https://github.com/gallery-so/gallery/assets/49758803/a08e8434-2aa0-46e0-8246-8bbb23338fe5

### Demo (dark mode)

https://github.com/gallery-so/gallery/assets/49758803/c416035a-f859-4e7c-9dd3-7311a5022013

### Other Improvement 
| Before | After |
|--------|--------|
| Screenshot 2023-12-07 at 6 56 11 AM | Screenshot 2023-12-05 at 11 41 27 PM |

### Edge Cases

List any common edge cases that you have considered and tested.

### Testing Steps

Provide steps on how the reviewer can test the changes.

### Checklist

Please make sure to review and check all of the following:

- [x] I've tested the changes and all tests pass.
- [ ] (if web) I've tested the changes on various desktop screen sizes to ensure responsiveness.
- [x] (if mobile) I've tested the changes on both light and dark modes.

### CODE REVIEW

File: apps/mobile/src/screens/PostScreen/MiniPostOpenGraphPreview.tsx [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+  return (
+    <View className="flex w-[358]px h-[188px] bg-[#F9F9F9] items-center justify-center">
+      <View className="flex flex-row items-center space-x-5">
+        <OpenBracket />
+        <View className="flex flex-row space-x-4">
+          <ImagePreview tokenUrl={imageUrl} onError={noop} />
+          <View className="flex flex-col justify-center">
+            <View className="flex flex-row items-center space-x-1">
+              {profileImageUrl ? (
+                <ImagePreview isPfp tokenUrl={profileImageUrl ?? ''} onError={noop} />
```
Comment (Rohan-cp) [inline apps/mobile/src/screens/PostScreen/MiniPostOpenGraphPreview.tsx:45]: not sure how to error handle with `onError` for the ImagePreview here and on L41

File: apps/mobile/src/screens/PostScreen/SharePostBottomSheet.tsx [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+              icon={btnDetails.icon}
+              onPress={() => handleShareButtonPress(btnDetails.baseComposePostUrl)}
+            />
+          ))}
+        </View>
+
+        <View className="flex flex-row">
+          <View className="w-9/12 mr-2">
+            <FadedInput
+              textClassName="text-[#707070] h-[24px]"
```
Comment (jakzaizzat) [inline apps/mobile/src/screens/PostScreen/SharePostBottomSheet.tsx:242]: nit: can change too `text-metal h-6`

File: apps/mobile/src/components/FadedInput.tsx [hunk 3]
```diff
@@ -4,14 +4,22 @@ import { TextInput, TextInputProps, View } from 'react-native';
 
 import colors from '~/shared/theme/colors';
 
-type Props = TextInputProps & { icon?: ReactNode };
+type Props = TextInputProps & { icon?: ReactNode; editable?: boolean; textClassName?: string };
```
Comment (jakzaizzat) [inline apps/mobile/src/components/FadedInput.tsx:7]: If Im not mistaken, the `editable` props already exist inside the `TextInputProps`

File: apps/mobile/src/screens/PostScreen/MiniPostOpenGraphPreview.tsx [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+      eventElementId="ProfilePicture"
+      eventName="ProfilePicture pressed"
+      eventContext={null}
+      size="xxs"
+      letter={letter ?? '?'}
+    />
+  );
+
+  return (
+    <View className="flex w-[358]px h-[188px] bg-[#F9F9F9] items-center justify-center">
```
Comment (jakzaizzat) [inline apps/mobile/src/screens/PostScreen/MiniPostOpenGraphPreview.tsx:36]: Update to `bg-offWhite`

File: apps/mobile/src/screens/PostScreen/MiniPostOpenGraphPreview.tsx (hunk 1 above)
Comment (jakzaizzat) [reply apps/mobile/src/screens/PostScreen/MiniPostOpenGraphPreview.tsx:44]: Maybe can mark it as failed, just like in `NftPreviewAsset.tsx`

```tsx
  const handleError = useCallback(() => {
    markTokenAsFailed(
      token.dbid,
      new CouldNotRenderNftError('RawNftPreviewAsset', 'Failed to render', { id: token.dbid })
    );
  }, [markTokenAsFailed, token.dbid]);
```

File: apps/mobile/src/screens/PostScreen/SharePostBottomSheet.tsx [hunk 5]
```diff
@@ -0,0 +1,294 @@
...
+    `,
+    { postId: postId }
+  );
+
+  const { post } = queryResponse;
+  const { bottom } = useSafeAreaPadding();
+  const [hasCopiedUrl, setHasCopiedUrl] = useState(false);
+
+  const bottomSheetRef = useRef<GalleryBottomSheetModalType>(null);
+  useEffect(() => bottomSheetRef.current?.present(), []);
```
Comment (jakzaizzat) [inline apps/mobile/src/screens/PostScreen/SharePostBottomSheet.tsx:83]: Can we used `forwardRef` for this bottom sheet component? So, we don't the useEffect and we can control (open/close) it from the parent component. 

If we wanna go with the forwardRef, we might need to move the Suspense inside this bottom sheet. 

There is an example how we used it at `WarningPostBottomSheet.tsx`

**Pull request merged**